core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -10,9 +10,39 @@ import scala.collection.JavaConverters.{asScalaBufferConverter, collectionAsScal
 
 class TSqlExpressionBuilder extends TSqlParserBaseVisitor[ir.Expression] with ParserCommon {
 
-  // TODO: A lot of work here for things that are not just simple x.y.z
-  override def visitSelectListElem(ctx: TSqlParser.SelectListElemContext): ir.Expression =
-    ctx.expressionElem.accept(this)
+  override def visitSelectListElem(ctx: TSqlParser.SelectListElemContext): ir.Expression = {
+    val handlers = Seq(
+      // TODO: asterisk not fully handled
+      Option(ctx.asterisk()).map(_ => ctx.asterisk().accept(this)),
+      // TODO: UDT elements seem broken in the grammar
+      Option(ctx.udtElem()).map(_ => ctx.udtElem().accept(this)),
+      Option(ctx.LOCAL_ID()).map(_ => buildLocalAssign(ctx)),
+      Option(ctx.expressionElem()).map(_ => ctx.expressionElem().accept(this)))
+    handlers.flatten.headOption.getOrElse(ir.UnresolvedExpression("Unsupported SelectListElem"))
+  }
+
+  /**
+   * Build a local variable assignment from a column source
+   *
+   * @param ctx
+   *   the parse tree containing the assignment
+   */
+  private def buildLocalAssign(ctx: TSqlParser.SelectListElemContext): ir.Expression = {
+    val localId = ir.Identifier(ctx.LOCAL_ID().getText, isQuoted = false)
+    val expression = ctx.expression().accept(this)
+    ctx.op.getType match {
+      case EQ => ir.Assign(localId, expression)
+      case PE => ir.Assign(localId, ir.Add(localId, expression))
+      case ME => ir.Assign(localId, ir.Subtract(localId, expression))
+      case SE => ir.Assign(localId, ir.Multiply(localId, expression))
+      case DE => ir.Assign(localId, ir.Divide(localId, expression))
+      case MEA => ir.Assign(localId, ir.Mod(localId, expression))
+      case AND_ASSIGN => ir.Assign(localId, ir.BitwiseAnd(localId, expression))
+      case OR_ASSIGN => ir.Assign(localId, ir.BitwiseOr(localId, expression))
+      case XOR_ASSIGN => ir.Assign(localId, ir.BitwiseXor(localId, expression))
+      case _ => ir.UnresolvedExpression(ctx.getText) // Handle unexpected operation types
+    }
+  }
```

Comment:
Same remark as above: this could be fully covered using mockito

Reply:
Ah - I see what you mean. Force line 43 to be executed. However, it can't really be reached, so I am a bit loathe to spend time on mockito until it definitely gives some benefit.